extension/jaegerremotesampling/extension.go

```diff
@@ -60,15 +59,12 @@ func (jrse *jrsExtension) Start(ctx context.Context, host component.Host) error
 	}
 
 	if jrse.cfg.Source.Remote != nil {
-		conn, err := jrse.cfg.Source.Remote.ToClientConn(ctx, host, jrse.telemetry)
+		remoteStore, closer, err := internal.NewRemoteStrategyStore(ctx, host, jrse.telemetry, jrse.cfg.Source.Remote)
```

Comment:
Note for reviewers: the before-state of this code was using an imported library to construct a `StrategyStore` (`grpcStore.NewConfigManager(conn)`). Because I took a composition/delegation approach to enhance outgoing gRPC context with headers, and that's a bit more code than a single call to an imported type, I defined it in `remote_strategy_store.go` rather than further expand `extension.go`.

The functionality of the extracted type is still covered by `extension_test` (an integration test) and that test was actually deepened to cover performed outbound gRPC calls when using remote mode (showing the intended behavior after this PR).

Reply:
Sounds valid to me :+1: